src/kaocha/config.clj

```diff
@@ -217,6 +222,39 @@
                           tests)))))
     config))
 
+(defn load-config2
+  "Loads config from config-file, factoring in profile specified using profile,
+  and displaying messages about any errors."
+  ([config-file]
+   (load-config2 config-file nil {} nil nil))
+  ([config-file profile opts]
+   (load-config2 config-file profile opts nil nil))
+  ([config-file profile opts cli-options cli-args]
+   (let [config (cond-> config-file
+                    true (load-config-file (if profile (assoc opts :profile profile) opts))
+                    cli-options (apply-cli-opts cli-options)
+                    cli-args (apply-cli-args cli-args))
+
+         check_config_file (when (not (. (io/file (or config-file "tests.edn")) exists))
```

Comment:
Dashes and `(.exists ...)` would be more idiomatic here. The `(. ... foo)` syntax is really only used in Clojure core code while bootstrapping.